validate/validate_test.go

```diff
@@ -63,6 +63,40 @@ func TestJSONSchema(t *testing.T) {
 		},
 		{
 			config: &rspec.Spec{
+				Version: "1.0.0",
+				Linux:   &rspec.Linux{},
+			},
+			error: "",
+		},
+		{
+			config: &rspec.Spec{
+				Version: "1.0.0",
+				Linux: &rspec.Linux{
+					RootfsPropagation: "",
```

Comment:
"" is not valid value for RootfsPropagation why error is ""?

Reply:
> "" is not valid value for RootfsPropagation why error is ""?

I exoect because it's `omitempty`, although I haven't tracked that down to specific gojsonschema code.

Reply:
I think omitempty  means it can be unset but not ""

Reply:
> I think omitempty means it can be unset but not ""

For a non-pointer `omitempty` like [`RootfsPropagation`][1], an empty string is as close as you can get to unset.

[1]: https://github.com/opencontainers/runtime-spec/blob/v1.0.0/specs-go/config.go#L154

Reply:
I think the `RootfsPropagation: "" `test should be deleted.

Reply:
> I think the `RootfsPropagation: ""` test should be deleted.

Why?  With the current non-pointer property, that's how you ex[licitly unset the property.  I think we need to be sure clearing `RootfsPropagation` is legal, because as this thread demonstrates, it's not immediately obvious.

Reply:
If it is not set, its value is also `""`, which is repeated with the previous test (without setting the value), and I think the test of this value is meaningless.

Reply:
> If it is not set, its value is also "", which is repeated with the previous test (without setting the value)...

Yeah, but it doesn't hurt to double check, and the explicit-empty-string case surprised @Mashimiao [at first][1].  If test rephrasings help expose surprises, O=I think we want them.

[1]: https://github.com/opencontainers/runtime-tools/pull/476#discussion_r139081493